libbeat/processors/extract_array/extract_array_test.go

```diff
@@ -0,0 +1,216 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package extract_array
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+func TestExtractArrayProcessor_String(t *testing.T) {
+	p, err := NewExtractArray(common.MustNewConfigFrom(common.MapStr{
+		"field": "csv",
+		"mappings": common.MapStr{
+			"source.ip":         0,
+			"network.transport": 2,
+			"destination.ip":    99,
+		},
+	}))
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, "extract_array={field=csv, mappings=[{0 source.ip} {2 network.transport} {99 destination.ip}]}", p.String())
+}
+
+func TestExtractArrayProcessor_Run(t *testing.T) {
+	tests := map[string]struct {
+		config   common.MapStr
+		input    beat.Event
+		expected beat.Event
+		fail     bool
+		afterFn  func(e *beat.Event)
+	}{
+		"sample": {
+			config: common.MapStr{
+				"field": "array",
+				"mappings": common.MapStr{
+					"dest.one": 1,
+					"dest.two": 2,
+				},
+			},
+			input: beat.Event{
+				Fields: common.MapStr{
+					"array": []interface{}{"zero", 1, common.MapStr{"two": 2}},
+				},
+			},
+			expected: beat.Event{
+				Fields: common.MapStr{
+					"array":    []interface{}{"zero", 1, common.MapStr{"two": 2}},
+					"dest.one": 1,
+					"dest.two": common.MapStr{"two": 2},
+				},
+			},
+		},
+
+		"modified elements": {
+			config: common.MapStr{
+				"field": "array",
+				"mappings": common.MapStr{
+					"dest.one": 1,
+					"dest.two": 2,
+				},
+			},
+			input: beat.Event{
+				Fields: common.MapStr{
+					"array": []interface{}{"zero", 1, common.MapStr{"two": 2}},
+				},
+			},
+			expected: beat.Event{
+				Fields: common.MapStr{
+					"array":    []interface{}{"zero", 1, common.MapStr{"two": 2}},
+					"dest.one": 1,
+					"dest.two": common.MapStr{"two": 3},
+				},
+			},
+			afterFn: func(e *beat.Event) {
+				e.PutValue("dest.two.two", 3)
+			},
```

Comment:
What is it testing?

Reply:
This checks that after extracting an object from an array, modifying either the original or copied object doesn't affect the other.